lib/schema_tools/modules/object_properties.rb

```diff
@@ -0,0 +1,32 @@
+module SchemaTools
+  module Modules
+    module ObjectProperties
+      PROPERTIES = 'properties'
+      ALLOF = 'allOf'
+      ANYOF = 'anyOf'
+      ONEOF = 'oneOf'
+      PROPERTY_CONTAINERS = [ PROPERTIES, ALLOF, ANYOF, ONEOF ]
+
+      def all_properties(schema_hash, properties_hash = ActiveSupport::HashWithIndifferentAccess.new)
```

Comment:
It isnt, I was thinking it may be useful for appending properties to (and originally was using it in place of `merge!`)
I can remove if you prefer